lib/puppet/network/http/api/indirected_routes.rb

```diff
@@ -190,9 +190,21 @@ def accepted_response_formatter_or_json_for(model_class, request)
 
   def read_body_into_model(model_class, request)
     data = request.body.to_s
+    formatter = request.formatter
+
+    if formatter && formatter.supported?(model_class)
+      begin
+        return model_class.convert_from(formatter.name.to_s, data)
+      rescue => e
+        raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(
+          _("The request body is invalid: %{message}") % { message: e.message })
+      end
+    end
 
-    format = request.format
-    model_class.convert_from(format, data)
+    #TRANSLATORS "mime-type" is a keyword and should not be translated
+    raise Puppet::Network::HTTP::Error::HTTPUnsupportedMediaTypeError.new(
+      _("Client sent a mime-type (%{header}) that doesn't correspond to a format we support") % { header: request.headers['content-type'] },
+      Puppet::Network::HTTP::Issues::UNSUPPORTED_MEDIA_TYPE)
```

Comment:
This is probably a crazy question, because why would the answer possibly be yes, but is there any other kind of request we'd serve besides `PUT` / `save` where we'd expect/use the request's `Content-Type`? I only ask because...well you know. We do unusual things sometimes.